pkg/chunk/chunk.go

```diff
@@ -282,33 +270,27 @@ func (c *Chunk) Decode(decodeContext *DecodeContext, input []byte) error {
 	if err := binary.Read(r, binary.BigEndian, &metadataLen); err != nil {
 		return err
 	}
-	var tempMetadata struct {
-		Chunk
-		RawMetric json.RawMessage `json:"metric"` // Override to defer parsing
-	}
-	decodeContext.reader.Reset(&io.LimitedReader{
-		N: int64(metadataLen),
-		R: r,
-	})
+	var tempMetadata Chunk
+	decodeContext.reader.Reset(r)
+	json := jsoniter.ConfigFastest
 	err := json.NewDecoder(decodeContext.reader).Decode(&tempMetadata)
 	if err != nil {
 		return err
 	}
+	if len(input)-r.Len() != int(metadataLen) {
```

Comment:
Just to make sure I am understanding correctly, it looks like we used to ignore data after metadataLen, now we throw an error? If so, have you seen this error happen/what are the ramifications of the change?

Reply:
Previously, if the stored `metadataLength` was less than the actual length of the compressed json the `LimitedReader` would truncate it, hence cause a decoding error; if it was too high then this would have no effect.

I added this check because I felt bad about simply deleting the `LimitedReader` and hence never looking at `metadataLen`.  I think it's about the same as the other checks hereabouts - if anything seems off then we drop the chunk and report an error.

Note the stored quantity is actually 4 bytes higher than the amount we read; I added a comment about that.

Reply:
👍 Plus this error message will be more clear than a decode error. Thanks for that comment as well, it was helpful looking through this.